### ISSUE TITLE
Revisit using the relay parent of a candidate for ordering in the dispute coordinator

### ISSUE THREAD

**Issue by eskimor** [paritytech/polkadot#5007]
As multiple candidates will be able to pick the same relay parent and a parachain can even choose (within bounds) the relay parent. It is not clear how much of a problem that really is, but the underlying assumption that a candidate with an older relay parent, must be older than a candidate of a different parachain with a newer relay parent, clearly no longer holds.

Given that the range of pickable relay parents is limited, this might still be acceptable, but it just became more fragile. Maybe going back to the including block number would be better and we should just handle the ambiguity in some sane fashion.

**Comment by rphmeier** [comment]
I suppose this would mean ordering based more so on the relay-block number where the block is included.

**Comment by rphmeier** [comment]
dumb question: why don't we do that already?

**Comment by eskimor** [comment]
We used to, but you rightly pointed out that there is not a single including block in the case of forks. So we can't use the block number of the including block for unambiguous ordering.

**Comment by rphmeier** [comment]
The changes in the relay chain only relax the minimum age of the relay-parent and don't remove the restriction altogether. So the condition is going from "a block with later relay parent is definitely newer than one with an earlier included at the same time" to something like "a block with later relay parent is at most 30 seconds older than a block with an earlier relay parent included at the same time". Which doesn't seem material for security since our unit of time is hours, not seconds.

**Comment by eskimor** [comment]
Correct, we just come short of encoding the actual content and instead use a proxy, which makes it more brittle. I agree, that with reasonable restrictions on the relay parent a parachain can pick, we should be "fine". In any case, I would like to keep this open for now, because depending on what we will actually end up with, we might be able to improve this. E.g. if we have the used depth available, maybe we can take advantage of that.

**Comment by burdges** [comment]
Interesting, we'll delay XCMP messages until the sender gets included I guess.  We (almost?) needed to do this anyways, so no problem.
